addons/pyroscope/readme.md

```diff
@@ -0,0 +1,233 @@
+# Pyroscope
+
+This addon is built based [Pyroscope](https://github.com/pyroscope-io/pyroscope), which consist of server and agent. It allows the user to collect, store, and query the profiling data in a CPU and disk efficient way.
+
+## install
+
+```shell
+vela addon enable pyroscope
+```
+After enable pyroscope successfully, you can execute command to expose the port for Dashboard UI
+```shell
+export POD_NAME=$(kubectl get pods --namespace vela-system -l "app.kubernetes.io/name=pyroscope,app.kubernetes.io/instance=pyroscope" -o jsonpath="{.items[0].metadata.name}")
+export CONTAINER_PORT=$(kubectl get pod --namespace vela-system $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+kubectl --namespace vela-system port-forward $POD_NAME 8083:$CONTAINER_PORT
+```
+
+## uninstall
+
+```shell
+vela addon disable pyroscope
+```
+
+
+# How to start
+Use a component typed webservice to start, keep the following to pyroscope-demo.yaml, then vela up -f app-demo.yaml
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: pyroscope-app
+  namespace: fourier
+spec:
+  components:
+    - name: pyroscope-comp-01
+      type: webservice
+      properties:
+        image: nginx:latest
+        ports:
+          - expose: true
+            port: 80
+            protocol: TCP
+        imagePullPolicy: IfNotPresent
+      traits:
+        - type: pyroscope
+          properties:
+            server: "http://pyroscope-server:9084"
```

Comment:
does every cluster need such a sever when we specify the trait? How it works in multi-cluster delivery case？

Reply:
Multi-cluster client has the ability to push data to one server, user should manually publish domian and create ingress.